pkg/controllers/provisioning/v1alpha1/reallocator/controller.go

```diff
@@ -53,31 +55,60 @@ func (c *Controller) Name() string {
 func NewController(kubeClient client.Client, cloudProvider cloudprovider.Factory) *Controller {
 	return &Controller{
 		filter:        &Filter{kubeClient: kubeClient},
+		collector:     &Collector{kubeClient: kubeClient},
 		cloudProvider: cloudProvider,
 	}
 }
 
 // Reconcile executes an allocation control loop for the resource
 func (c *Controller) Reconcile(object controllers.Object) error {
+	provisioner := object.(*v1alpha1.Provisioner)
 	ctx := context.TODO()
-	// 1. Filter all nodes with Karpenter TTL label
-	// TODO: Filter only nodes with the Provisioner label
 
-	// 2. Remove TTL from nodes with TTL label that have pods and if past TTL, cordon/drain node
+	// 1. Filter all nodes with Karpenter labels
+	ttlNodes, err := c.filter.getNodesWithLabels(ctx, []string{
+		provisioning.ProvisionerNamespaceLabelKey,
+		provisioning.ProvisionerNameLabelKey,
+	})
+	if err != nil {
+		return fmt.Errorf("listing ttl provisioner nodes, %w", err)
```

Comment:
This error message is a little incoherent. Think of how this will all wire together and how it'll read to the debugger.

"listing ttled nodes" is minimally better english. Seems fine if we want a concept of "TTLed". Open to suggestions if you think we can do better on that naming, but we should use the concept consistently.

Reply:
I think TTLed is the best way to describe it succinctly